### PR TITLE
Remove redundant pretty calls, fix various `print` calls missing the `io` argument

### PR DESCRIPTION
**Pull request by fingolfin**
Also fix `Base.show(io::IO, S::MPolyLeadingMonOne)` method to set
`supercompact` to `true` *after* checking whether it is equal to `true`.

### CODE REVIEW

**Review by lgoettgens — APPROVED** [review]

**Comment by thofma** [comment]
doctests broken?

File: src/Rings/mpolyquo-localizations.jl [hunk 1]
```diff
@@ -219,7 +219,6 @@ function Base.show(io::IO, ::MIME"text/plain", L::MPolyQuoLocRing)
 end
 
 function Base.show(io::IO, L::MPolyQuoLocRing)
-  io = pretty(io)
   if get(io, :supercompact, false)
     print(io, "Localized quotient of multivariate polynomial ring")
   else
```
Comment (lgoettgens) [inline src/Rings/mpolyquo-localizations.jl:226]: RIght below this line, there is a similar thing wrong with the order of calling `pretty` and setting `supercompact => true` as the one already fixed for `MPolyLeadingMonOne`. The doctest failure seems related to this.

Comment (fingolfin) [reply src/Rings/mpolyquo-localizations.jl:226]: I fixed the doctest, but I couldn't find the spot you are talking about (I think?). If you still see an issue, please pinpoint me. Thanks!

Comment (lgoettgens) [reply src/Rings/mpolyquo-localizations.jl:226]: You already fixed it

**Comment by fingolfin** [comment]
Running into a bug in AA (see https://github.com/Nemocas/AbstractAlgebra.jl/pull/1447) will try to work around it.

**Pull request merged**